### ISSUE TITLE
filter-abund.py returns AttributeError when using --gzip flag

### ISSUE THREAD

**Issue by jvollme** [dib-lab/khmer#1308]
Hi,
I've just upgraded to khmer version 2.0 and was very happy about the ability of each pipeline component to read from and write to gzipped files.
This seems to work perfectly for "interleave-reads.py" and "normalize-by-median.py", however "filter-abund.py" aborts with the following error message:

Traceback (most recent call last):
  File "/work/home/jov14/tools/khmerEnv/bin/filter-abund.py", line 177, in <module>
    main()
  File "/work/home/jov14/tools/khmerEnv/bin/filter-abund.py", line 156, in main
    outfp = get_file_writer(outfile, args.gzip, args.bzip)
  File "/work/home/jov14/tools/khmerEnv/lib/python2.7/site-packages/khmer/kfile.py", line 236, in get_file_writer
    ofile = gzip.GzipFile(fileobj=file_handle, mode='w')
  File "/usr/local/lib/python2.7/gzip.py", line 136, in __init__
    self._write_gzip_header()
  File "/usr/local/lib/python2.7/gzip.py", line 165, in _write_gzip_header
    self.fileobj.write('\037\213')             # magic header
AttributeError: 'str' object has no attribute 'write

However it seems to work fine if I leave the --gzip flag away and just produce uncompressed results (that i can gzip myself)

**Comment by ctb** [comment]
On Fri, Sep 18, 2015 at 02:43:14AM -0700, JohnVollmers wrote:
> Hi,
> I've just upgraded to khmer version 2.0 and was very happy about the ability of each pipeline component to read from and write to gzipped files.
> This seems to work perfectly for "interleave-reads.py" and "normalize-by-median.py", however "filter-abund.py" aborts with the following error message:
> 
> Traceback (most recent call last):
>   File "/work/home/jov14/tools/khmerEnv/bin/filter-abund.py", line 177, in <module>
>     main()
>   File "/work/home/jov14/tools/khmerEnv/bin/filter-abund.py", line 156, in main
>     outfp = get_file_writer(outfile, args.gzip, args.bzip)
>   File "/work/home/jov14/tools/khmerEnv/lib/python2.7/site-packages/khmer/kfile.py", line 236, in get_file_writer
>     ofile = gzip.GzipFile(fileobj=file_handle, mode='w')
>   File "/usr/local/lib/python2.7/gzip.py", line 136, in __init__
>     self._write_gzip_header()
>   File "/usr/local/lib/python2.7/gzip.py", line 165, in _write_gzip_header
>     self.fileobj.write('\037\213')             # magic header
> AttributeError: 'str' object has no attribute 'write
> 
> However it seems to work fine if I leave the --gzip flag away and just produce uncompressed results (that i can gzip myself)

Thanks for reporting this - sounds like a bug!  If you can give us the
command line you're using it might help when we're trying to reproduce it
(although I think I can guess ;).

(Note that you can use gzip in a pipe with 'gzip -c'.)

**Comment by ctb** [comment]
Reproduced:

```
scripts/load-into-counting.py kmers.g data/100k-filtered.fa
cat data/100k-filtered.fa | scripts/filter-abund.py kmers.g - -o xxx --gzip
```

yields
```
Traceback (most recent call last):
  File "scripts/filter-abund.py", line 177, in <module>
    main()
  File "scripts/filter-abund.py", line 156, in main
    outfp = get_file_writer(outfile, args.gzip, args.bzip)
  File "/Users/t/dev/khmer/khmer/kfile.py", line 236, in get_file_writer
    ofile = gzip.GzipFile(fileobj=file_handle, mode='w')
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/gzip.py", line 127, in __init__
    self._write_gzip_header()
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/gzip.py", line 156, in _write_gzip_header
    self.fileobj.write('\037\213')             # magic header
AttributeError: 'str' object has no attribute 'write'
```

**Comment by ctb** [comment]
Fixed in #1310.
